streams/src/main/java/org/apache/kafka/streams/processor/internals/StateRestorer.java

```diff
@@ -26,13 +26,13 @@
 
     static final int NO_CHECKPOINT = -1;
 
-    private final Long checkpoint;
```

Comment:
I moved the logic of casting nullable Long to long at the constructor so this can be long now. Also since we may want to update the value in case of re-initialization, it cannot be final anymore.